src/libpmem2/config.c

```diff
@@ -209,7 +209,7 @@ pmem2_config_set_address(struct pmem2_config *cfg, void *addr,
 		return PMEM2_E_INVALID_ADDRESS_REQUEST_TYPE;
 	}
 
-	if (request_type == PMEM2_ADDRESS_FIXED_NOREPLACE && !addr) {
+	if (!addr) {
```

Comment:
I'm not sure - this assumes that this enum will not be extended. I'd prefer a switch.

Reply:
Even if enum would be extended then the logic from a previous conditional expression must be somehow updated - but I will not insist I drop this PR.